### PR TITLE
Refactor HTTP API routes

### PR DESCRIPTION
**Pull request by xicombd**
Use dependency injection on routes so that we can start and stop the http-api multiple times without issues on the tests. 
~~Updated `jsipfs version` to work with the api running and added a test to make sure it works :)~~ Got similar bug to #70

### CODE REVIEW

File: src/http-api/index.js [hunk 1]
```diff
@@ -46,7 +46,7 @@ exports.start = (callback) => {
     server.connection({ host: gateway[2], port: gateway[4], labels: 'Gateway' })
 
     // load routes
-    require('./routes')
+    require('./routes')(server.select('API'))
```
Comment (diasdavid) [inline src/http-api/index.js]: pass the server object completely and let the routes decide where to attach

Comment (xicombd) [inline src/http-api/index.js]: @diasdavid done!

**Comment by xicombd** [comment]
Added the ipfs version test with the daemon running again, now fixed

**Comment by diasdavid** [comment]
thank you yet again @xicombd :D :D

**Pull request merged**